components/browser/menu/src/main/java/mozilla/components/browser/menu/WebExtensionBrowserMenu.kt

```diff
@@ -0,0 +1,107 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.menu
+
+import android.view.View
+import android.widget.PopupWindow
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.collect
+import mozilla.components.browser.menu.item.WebExtensionBrowserMenuItem
+import mozilla.components.browser.state.selector.selectedTab
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.SessionState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.webextension.BrowserAction
+import mozilla.components.lib.state.ext.flowScoped
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
+import java.util.concurrent.ConcurrentHashMap
+
+typealias WebExtensionBrowserAction = BrowserAction
+
+/**
+ * A popup menu for [WebExtensionBrowserAction] menu items.
+ */
+class WebExtensionBrowserMenu internal constructor(
+    adapter: BrowserMenuAdapter,
+    private val store: BrowserStore
+) : BrowserMenu(adapter) {
+    private var scope: CoroutineScope? = null
+
+    @kotlinx.coroutines.ExperimentalCoroutinesApi
+    override fun show(
+        anchor: View,
+        orientation: Orientation,
+        endOfMenuAlwaysVisible: Boolean,
+        onDismiss: () -> Unit
+    ): PopupWindow {
+        scope = store.flowScoped { flow ->
```

Comment:
@psymoon Could we find a more generic alternative without having to create a custom `BrowserMenu`? I think there are similar needs to this one, other developers may want to add dynamically items to the menu as well, but this solution only works for `WebExtensions`. What do you think?

Reply:
`menuItems` is an immutable list by design so I am trying to follow the same pattern. We don't dynamically add new items to the menu - just update tab specific browser action overrides that are already in the menu (for add-on live update while the menu is opened like in fennac). I think the use case here is niche enough for a custom `BrowserMenu`.

AFAIK, @NotWoods is working on a long term solution for menu [here](https://github.com/mozilla-mobile/android-components/pull/5496)

Reply:
What I'm a bit concern is the case when we want to add live updates to other items, which approuch should we take? Are we going to create a new type of   `BrowserMenu` that includes web exts and the new items? For moment, this solution is going work but it will be hard to scale it. If we don't other short term alternatives, I'm OK with keeping this one and refactor it later with a solution like  @NotWoods is proposing.

Reply:
@Amejia481 I'm just updating the object references and invalidating the respective views so you could technically accomplish that with current `BrowserMenu` implementation as well.

Reply:
A lighter approach could be adding two callback to`BrowserMenu` on  `show` and another to the `onDismissListener`. This way we could have a  `WebExtensionBrowserMenuFeature`, that gets the `menu` and starts listing for changes on `onShow` and stops listing when `onDismissListener`. Exactly as you are doing here. What do you think?

Reply:
@Amejia481 I think it is a viable concern and we should address it in the near future, especially when @NotWoods is ready to land his refactor. For now, the priority is to change as little as possible and easy for us to integrate.

Reply:
👍